### PR TITLE
Merge sort

### PR DESCRIPTION
**Pull request by tesmonrd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,8 @@
 # advanced-data-structures
+[![Build Status](https://travis-ci.org/paulsheridan/advanced-data-structures.svg?branch=hashtable)](https://travis-ci.org/paulsheridan/advanced-data-structures)
+
 A new data structures repo, brimming with some higher-level data structures.
+
```
Comment (nhuntwalker) [inline README.md:5]: You need to add information about your sorting algorithms

File: src/sort.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import random
+
+
+def insertionsort(in_list):
```
Comment (nhuntwalker) [inline src/sort.py:5]: Doc string

File: src/test_sort.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import random
+
+
+EDGE_CASES = [[[100, 100, 100, 100, 100], [100, 100, 100, 100, 100]],
+              [[999999999999999, 0], [0, 999999999999999]],
+              [[1000000000, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 1000000000]],
+              [[], []],
+              [[1, 100, 1, 100, 1, 100, 1, 100, 1, 100],
+               [1, 1, 1, 1, 1, 100, 100, 100, 100, 100]]
+              ]
```
Comment (nhuntwalker) [inline src/test_sort.py]: Your edge cases should include sorting strings as well, or rather a list of strings.

Comment (nhuntwalker) [inline src/test_sort.py:11]: Your tests should include some negative numbers just to ensure that it works for more than just 0+. You should also include a list of strings.

File: src/sort.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    for idx in range(1, len(in_list)):
+        while idx > 0 and in_list[idx] < in_list[idx - 1]:
+            in_list[idx], in_list[idx - 1] = in_list[idx - 1], in_list[idx]
+            idx -= 1
+    return in_list
+
+
+def mergesort(in_list):
```
Comment (nhuntwalker) [inline src/sort.py:13]: Docstring

File: src/sort.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    if len(in_list) > 1:
+        mid = len(in_list)//2
+        left = mergesort(in_list[:mid])
+        right = mergesort(in_list[mid:])
+        return(merge(left, right))
+    else:
+        return in_list
+
+
+def merge(left, right):
```
Comment (nhuntwalker) [inline src/sort.py:23]: Docstring
